src/components/pages/Weather/Weather.js

```diff
@@ -1,11 +1,57 @@
 import React from 'react';
+import firebase from 'firebase/app';
 import './Weather.scss';
+import connection from '../../../helpers/data/connection';
+import getWeather2 from '../../../helpers/data/weatherRequests';
+import CurrentWeather from './CurrentWeather';
+import WeatherForm from './WeatherForm';
+
 
 class Weather extends React.Component {
+  state = {
+    uid: '',
+    weatherArray2: [],
+  }
+
+  getWx = (uid2) => {
+    getWeather2.getWeather(uid2)
+      .then((weatherArray2) => {
+        this.setState({ weatherArray2 });
+      })
+      .catch(err => console.error('error with getWeather', err));
+  }
+
+  componentDidMount() {
+    connection();
+    this.removeListener = firebase.auth().onAuthStateChanged((user) => {
```

Comment:
You don't need to replicate this function in this component // take this out. It's better off in the App component.

Reply:
You can use authRequests.getCurrentUid to get the uid here instead